sklearn/utils/tests/test_validation.py

```diff
@@ -500,4 +501,13 @@ def test_check_consistent_length():
     # Despite ensembles having __len__ they must raise TypeError
     assert_raises_regexp(TypeError, 'estimator', check_consistent_length,
                          [1, 2], RandomForestRegressor())
+
+    # check pandas dataframe with 'fit' column does not raise error
+    try:
+        import pandas as pd
+        X = np.array([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+        X_df = pd.DataFrame(X, columns=['a', 'b', 'fit'])
+        check_consistent_length(X_df)
+    except ImportError:
+        raise SkipTest("Pandas not found")
```

Comment:
Can you put your test in a separate test function please? Any lines after your test will not be run if pandas is not installed which is a bit dodgy.

Also add a link to the issue to add some context. 

To be honest maybe the simplest thing to do is to have a test that does not require pandas, i.e. something like:
```py
X = np.ones(5)
X.fit = 'an non-callable attribute'
check_consistent_length(X)
```